langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingModel.java

```diff
@@ -0,0 +1,180 @@
+ * <a href="https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-embed.html">here</a>.
+ */
+public class BedrockCohereEmbeddingModel extends AbstractBedrockEmbeddingModel<BedrockCohereEmbeddingResponse> {
+
+    private String model;
+    private String inputType;
+    private String truncate;
+    private String embeddingType;
+
+    public BedrockCohereEmbeddingModel(String model, String inputType, String truncate, String embeddingType,
+                                       AbstractBedrockEmbeddingModelBuilder builder) {
+        super(builder);
+        this.model = model;
+        this.inputType = inputType;
+        this.truncate = truncate;
+        this.embeddingType = embeddingType;
+    }
+
+    @Override
+    protected List<Map<String, Object>> getRequestParameters(List<TextSegment> textSegments) {
+        if (model == null) {
+            throw new IllegalArgumentException("Model is required.");
+        }
+        if (inputType == null) {
+            throw new IllegalArgumentException("Input type is required.");
+        }
+        if (!embeddingType.equalsIgnoreCase("float")) {
+            throw new IllegalArgumentException("Only float embedding type is supported.");
+        }
+
+        List<Map<String, Object>> result = new ArrayList<>();
+        for (TextSegment textSegment : textSegments) {
+            Map<String, Object> parameters = new HashMap<>();
+            parameters.put("texts", List.of(textSegment.text()));
+            parameters.put("input_type", inputType);
+            parameters.put("truncate", truncate);
+            parameters.put("embedding_types", List.of(embeddingType));
+            result.add(parameters);
+        }
+        return result;
+    }
+
+    @Override
+    protected Class<BedrockCohereEmbeddingResponse> getResponseClassType() {
+        return BedrockCohereEmbeddingResponse.class;
+    }
+
+    @Override
+    protected String getModelId() {
+        return model;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder extends AbstractBedrockEmbeddingModelBuilder<BedrockCohereEmbeddingResponse, BedrockCohereEmbeddingModel, Builder> {
+
+        private String model;
+        private String inputType;
+        private String truncate = Truncate.NONE.getValue();
+        private String embeddingType = EmbeddingType.FLOAT.getValue();
+
+        @Override
+        protected Builder self() {
+            return this;
+        }
+
+        public Builder model(String model) {
+            this.model = model;
+            return this;
+        }
+
+        public Builder inputType(String inputType) {
+            this.inputType = inputType;
+            return this;
+        }
+
+        public Builder truncate(String truncate) {
+            this.truncate = truncate;
+            return this;
+        }
+
+        public Builder embeddingType(String embeddingType) {
```

Comment:
```suggestion
        public Builder embeddingTypes(String embeddingTypes) {
```

To keep it consistent with the actual `embedding_types` request parameter.

But if only `float` is supported, I would either remove this setter completely or remove the validation. Probably the second option